libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -512,13 +512,13 @@ bool ContractCompiler::visit(FunctionDefinition const& _function)
 	if (!_function.isConstructor())
 		// adding 1 for return address.
 		m_context.adjustStackOffset(parametersSize + 1);
-	for (ASTPointer<VariableDeclaration const> const& variable: _function.parameters())
+	for (ASTPointer<VariableDeclaration const> const variable: _function.parameters())
```

Comment:
I tested this again with the original version and get this error
```C++
/home/marcuswentz/lllc/libsolidity/codegen/ContractCompiler.cpp: In member function ‘virtual bool dev::solidity::ContractCompiler::visit(const dev::solidity::FunctionDefinition&)’:
/home/marcuswentz/lllc/libsolidity/codegen/ContractCompiler.cpp:515:59: error: loop variable ‘variable’ of type ‘dev::solidity::ASTPointer<const dev::solidity::VariableDeclaration>&’ {aka ‘const std::shared_ptr<const dev::solidity::VariableDeclaration>&’} binds to a temporary constructed from type ‘const std::shared_ptr<dev::solidity::VariableDeclaration>’ [-Werror=range-loop-construct]
  515 |         for (ASTPointer<VariableDeclaration const> const& variable: _function.parameters())
      |                                                           ^~~~~~~~
/home/marcuswentz/lllc/libsolidity/codegen/ContractCompiler.cpp:515:59: note: use non-reference type ‘dev::solidity::ASTPointer<const dev::solidity::VariableDeclaration>’ {aka ‘const std::shared_ptr<const dev::solidity::VariableDeclaration>’} to make the copy explicit or ‘const std::shared_ptr<dev::solidity::VariableDeclaration>&’ to prevent copying
cc1plus: all warnings being treated as errors
make[2]: *** [libsolidity/CMakeFiles/solidity.dir/build.make:440: libsolidity/CMakeFiles/solidity.dir/codegen/ContractCompiler.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:481: libsolidity/CMakeFiles/solidity.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```
The error is gone when I remove "&"
```C++
for (ASTPointer<VariableDeclaration const> const variable: _function.parameters())
```